src/dataframerow/dataframerow.jl

```diff
@@ -133,57 +233,28 @@ function rowhash(cols::Tuple{Vararg{AbstractVector}}, r::Int, h::UInt = zero(UIn
 end
 
 Base.hash(r::DataFrameRow, h::UInt = zero(UInt)) =
-    rowhash(ntuple(i -> parent(r)[i], ncol(parent(r))), row(r), h)
-
-# comparison of DataFrame rows
-# only the rows of the same DataFrame could be compared
-# rows are equal if they have the same values (while the row indices could differ)
-# if all non-missing values are equal, but there are missings, returns missing
-Base.:(==)(r1::DataFrameRow, r2::DataFrameRow) = isequal(r1, r2)
+    rowhash(ntuple(col -> parent(r)[getparentcols(r, col)], length(r)), row(r), h)
 
-function Base.isequal(r1::DataFrameRow, r2::DataFrameRow)
-    isequal_row(parent(r1), row(r1), parent(r2), row(r2))
+function Base.:(==)(r1::DataFrameRow, r2::DataFrameRow)
+    names(r1) == names(r2) || return false
```

Comment:
Better check whether the `cols` fields are equal when `parent(r1) === parent(r2) && row(r1) == row(r2)` as a fast path, and only if parents differ check whether names are equal.

Reply:
Agreed.